### PR TITLE
ARM spec review for validateAddress for BillingRP

### PR DESCRIPTION
**Pull request by sarang-amondikar**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by adxsdknet** [comment]
# Automation for azure-sdk-for-net 
A PR has been created for you: 
https://github.com/azure/azure-sdk-for-net/pull/6319 
.NET SDK Commits: 
https://github.com/adxsdknet/azure-sdk-for-net/pull/6319/commits/71943203cdff3015f888ec561a3476b4fa82760e

**Review by kpajdzik — APPROVED** [review]

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json [hunk 1]
```diff
@@ -3233,6 +3280,39 @@
...
+        "$ref": "#/definitions/AddressValidationStatus"
+      },
+      "suggestedAddresses": {
+        "description": "list of suggested addresses.",
+        "type": "array",
+        "items": {
+          "$ref": "#/definitions/Address"
+        }
+      },
+      "validationMessage" : {
```
Comment (ravbhatnagar) [inline specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: why is validation error message part of the request body. If the validation on an address fails, shouldnt it return an error with the details part of the error body which complies with the ARM RPC error contract

**Review by ravbhatnagar — COMMENTED** [review]
just one minor question

File: specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json (hunk 1 above)
Comment (sarang-amondikar) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: Validation error message is not part of request body. If AV fails, it returns a message in ValidationMessage property which is a part of Response object.

Comment (ravbhatnagar) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: sorry i meant response body itself. What i meant is -> if POST /validateAddress fails to validate an address, shouldnt the response should be a failure? In the current model, the response will be a successful response, with the validation failure in the response body.

Comment (sarang-amondikar) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: oh, yes, it always responds with 200, we internally call into other service which always returns 200, with validation results in response body. We apply some business logic and return 200.

Comment (ravbhatnagar) [reply specification/billing/resource-manager/Microsoft.Billing/preview/2018-11-01-preview/billing.json]: ok, thanks for confirming.

**Pull request merged**